app/Http/Controllers/QuizAPIController.php

```diff
@@ -50,4 +50,18 @@ public function addNewQuiz(Request $request): JsonResponse
             ], 500);
         }
     }
+
+    public function displayQuizByID(int $id): JsonResponse {
+        try {
+             $singleQuiz = Quiz::with(['questions', 'questions.answers'])->find($id);
+        } catch (Exception $e) {
+            return response()->json([
+                "message" => "Quiz not found"
+            ], 404);
+        }
```

Comment:
The fault caught by `try {...} catch {...}` here is not that it didn't find a quiz.

What does this code do if you have 5 quizzes in your database, and then your frontend asks for quiz 200?

You are `try`ing a database operation that may fail. 'I crashed because the database was down' is a totally different thing to 'I didn't find a quiz'.